src/compiler/checker.ts

```diff
@@ -11303,6 +11323,15 @@ module ts {
             }
         }
 
+        function checkGrammarModuleElementContext(node: ModuleElement, errorMessage: DiagnosticMessage): boolean {
+            if (node.parent.kind !== SyntaxKind.SourceFile && node.parent.kind !== SyntaxKind.ModuleBlock && node.parent.kind !== SyntaxKind.ModuleDeclaration) {
+                grammarErrorOnFirstToken(node, errorMessage);

```

Comment:
I'll just switch this to fit with the original model where true means there was an error.